components/bpmn/org.wso2.carbon.bpmn.uuf.ui.stub/src/main/java/org.wso2.carbon.bpmn.uuf.ui/service/BPMNExplorerServiceImpl.java

```diff
@@ -0,0 +1,477 @@
+
+            if (isLoginSuccess.equals("true")) {
+                //retrieve session cookie with BPS
+                String str = response.getHeaders("Set-Cookie")[0].toString().split(";")[0];
+                cookie = str.split(" ")[1];
+            }
+        } catch (ClientProtocolException | IOException e) {
+            log.error("Exception : ", e);
+        }
+        return cookie;
+    }
+
+    /**
+     * Return BPMN process diagram data through BPMNDeploymentService
+     *
+     * @param processId
+     * @param sessionId
+     */
+    private Object getBPMNProcessDiagram(String processId, String sessionId) {
+        HttpPost httpPost;
+        String line;
+        StringEntity stringEntity;
+        String imageUrl = null;
+        try (CloseableHttpResponse response = null;
+             BufferedReader rd = null) {
+            httpPost = new HttpPost(
+                    baseUrl + "/services/BPMNDeploymentService.BPMNDeploymentServiceHttpsSoap11Endpoint/");
+            httpPost.addHeader(BPMNExplorerServiceConstants.COOKIE, sessionId);
+            httpPost.addHeader(BPMNExplorerServiceConstants.SOAP_ACTION, BPMNExplorerServiceConstants.SOAP_ACTION_GET_PROCESS_DIAGRAM );
+            httpPost.addHeader(BPMNExplorerServiceConstants.CONTENT_TYPE,  BPMNExplorerServiceConstants.XML_CONTENT_TYPE);
+
+            String imgPayload =
+                    "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" " +
+                            "xmlns:ser=\"http://services.mgt.core.bpmn.carbon.wso2" +
+                            ".org\"><soapenv:Header/><soapenv:Body><ser:getProcessDiagram><ser:processId>" +
+                            processId + "</ser:processId></ser:getProcessDiagram></soapenv:Body></soapenv:Envelope>";
+            stringEntity = new StringEntity(imgPayload, "UTF-8");
+            stringEntity.setChunked(true);
+
+            httpPost.setEntity(stringEntity);
+            response = client.execute(httpPost);
+            rd = new BufferedReader(
+                    new InputStreamReader(response.getEntity().getContent(), Charset.defaultCharset()));
+            StringBuffer imgResult = new StringBuffer();
+            while ((line = rd.readLine()) != null) {
+                imgResult.append(line);
+            }
+            String imageData = imgResult.toString().split(":return")[1];
+            imageUrl = "data:image/png;base64," + imageData.substring(1, imageData.length() - 4);
+
+        } catch (ClientProtocolException | IOException e) {
+            log.error("Exception : ", e);
+        }
+        return imageUrl;
+    }
+
+    /**
+     * Initialization of creating process instances
+     *
+     * @param processId
+     * @param host
+     * @param port
+     */
+    @Override
+    public JSONObject getFormData(String processId, String host, int port, String username, String password) throws JSONException, IOException {
```

Comment:
Most of the Java doc methods are missing the return annotation. Especially in public methods, it's better to have proper java documntation with all details.